ggml.c

```diff
@@ -5213,6 +5213,8 @@ struct ggml_tensor * ggml_view_tensor(
         result->nb[i] = src->nb[i];
     }
 
+    result->op = GGML_OP_VIEW;
```

Comment:
I added your proposal since it better expresses the KV cache dependency, until we figure out a better solution.

For this to work though, we need to set the `ggml_view_tensor` op to `GGML_OP_VIEW` as you have recently noted. I guess this change is fine

Reply:
In this case, we should also set `src[0]` to `src` in `ggml_view_tensor`, and add the additional dependency as `src[1]` instead in `llm_build_graph`. Otherwise, it would break old code that assumes that the source is in `src[0]`, such as this in `ggml-cuda`:
https://github.com/ggerganov/llama.cpp/blob/bd33e5ab92e7f214205792fc1cd9ca28e810f897/ggml-cuda.cu#L6539-L6544
`ggml_graph_import` may also be an issue, and I am not sure that it will import the second dependency in `src[1]` correctly, since it just creates a new tensor calling `ggml_view_4d`. Newer code should use `view_src` instead.

Reply:
Ah good points. I guess I'll come back to this later and make a better solution.
Made an issue so we don't forget: https://github.com/ggerganov/ggml/issues/502